src/pages/patientView/mutation/PatientViewMutationsDataStore.ts

```diff
@@ -0,0 +1,104 @@
+import {SimpleGetterLazyMobXTableApplicationDataStore} from "../../../shared/lib/ILazyMobXTableApplicationDataStore";
+import {Mutation} from "../../../shared/api/generated/CBioPortalAPI";
+import {action, computed, observable} from "mobx";
+import _ from "lodash";
+import ComplexKeySet from "../../../shared/lib/complexKeyDataStructures/ComplexKeySet";
+
+function mutationMatch(d:Mutation[], id:Mutation) {
+    return d[0].proteinChange === id.proteinChange &&
+        d[0].gene.hugoGeneSymbol === id.gene.hugoGeneSymbol;
+}
+
+function mutationIdKey(m:Mutation) {
+    return `{ "proteinChange": "${m.proteinChange}", "hugoGeneSymbol": "${m.gene.hugoGeneSymbol}" }`;
+}
+
+export default class PatientViewMutationsDataStore extends SimpleGetterLazyMobXTableApplicationDataStore<Mutation[]> {
+    @observable.ref private mouseOverMutation:Readonly<Mutation>|null = null;
+    private selectedMutationsMap = observable.map<Mutation>();
+    @observable private _onlyShowSelectedInTable = false;
+    @observable private _onlyShowSelectedInVAFChart = false;
+
+    public getMouseOverMutation() {
+        return this.mouseOverMutation;
+    }
+
+    public get onlyShowSelectedInTable() {
+        return this._onlyShowSelectedInTable;
+    }
+
+    public get onlyShowSelectedInVAFChart() {
+        return this._onlyShowSelectedInVAFChart;
+    }
+
+    public setMouseOverMutation(m:Readonly<Mutation>|null) {
+        this.mouseOverMutation = m;
+    }
+
+    public setOnlyShowSelectedInTable(o:boolean) {
+        this._onlyShowSelectedInTable = o;
+    }
+
+    public setOnlyShowSelectedInVAFChart(o:boolean) {
+        this._onlyShowSelectedInVAFChart = o;
+    }
+
+    @action
+    public toggleSelectedMutation(m:Readonly<Mutation>) {
+        const key = mutationIdKey(m);
+        if (this.selectedMutationsMap.has(key)) {
+            this.selectedMutationsMap.delete(key);
```

Comment:
would be interesting to discus whether we should be using maps like this or more immutable approach where we use lodash methods to replace entire reference.  i'm not totally sure.